modules/ppcp-axo/src/AxoModule.php

```diff
@@ -367,6 +369,60 @@ static function ( $methods ) {
 			}
 		);
 
+		// Trigger 3D Secure verification.
+		add_filter(
+			'ppcp_create_order_request_body_data',
+			function( array $data, string $payment_method ) use ( $c ): array {
+				if ( ! $c->get( 'axo.uk.enabled' ) ) {
+					return $data;
+				}
+
+				$logger = $c->get( 'woocommerce.logger.woocommerce' );
+				assert( $logger instanceof LoggerInterface );
+
+				if ( $payment_method !== AxoGateway::ID ) {
+					return $data;
+				}
+
+				if ( ! isset( $data['payment_source'] ) || ! isset( $data['payment_source']['card'] ) || ! is_object( $data['payment_source']['card'] ) ) {
+					return $data;
+				}
+
+				$settings_model = $c->get( 'settings.data.settings' );
+				assert( $settings_model instanceof SettingsModel );
+
+				$three_d_secure = $settings_model->get_three_d_secure_enum();
+				$card           = $data['payment_source']['card'];
+
+				if (
+					$three_d_secure === 'SCA_ALWAYS'
+					|| $three_d_secure === 'SCA_WHEN_REQUIRED'
+				) {
+					$card->attributes = array(
+						'verification' => array(
+							'method' => $three_d_secure,
+						),
+					);
+
+					$experience_context_builder = $c->get( 'wcgateway.builder.experience-context' );
+					assert( $experience_context_builder instanceof ExperienceContextBuilder );
+
+					$data['experience_context'] = $experience_context_builder
```

Comment:
I think it's set incorrectly? `experience_context` is supposed to be in `payment_source.*.experience_context`.

Also maybe it can be added where the order creation call is triggered initially (`AxoGateway`?)? hm, actually why this `ppcp_create_order_request_body_data` override is needed at all if it's only for `AxoGateway`? Wouldn't it be simpler to just put it in the `AxoGateway`?

Also if it's always only `card` then the API docs say it supports only cancel/return URLs.

Reply:
Thanks for the feedback, again @AlexP11223 :) Much cleaner approach - appreciate your input! 🙏

The Fastlane 3DS docs show `experience_context` at the root level, not nested under `payment_source.card`.

I have kept it since it provides the return/cancel URLs for authentication redirects.